client/extensions/woocommerce/state/action-list/README.md

```diff
@@ -20,107 +20,115 @@ created.)
 
 ## What is it?
 
-The Action List is a declarative JavaScript array (no functions or anything else
-that is not serializable). Structurally, it's a single array of "steps",
-which are objects that describe each sequential step in the list. Using the above
-example, this could look like:
+The action list is a structure that defines the steps needed to be executed.
+It is passed from one action to the next until it is all complete.
+The form of the action list changes to denote current state as it is passed on.
 
+### Lifecycle
+
+The action list starts out with a list of steps:
 ```
 {
-  steps: [
-    { description: "Creating product: super-awesome thing", action: { ... } },
-    { description: "Creating super-awesome thing variation: small", action: { ... } },
-    { description: "Creating super-awesome thing variation: large", action: { ... } }
-  ]
-  successAction: { type: 'MY_SUCCESS_ACTION' },
-  failureAction: { type: 'MY_FAILURE_ACTION' },
-  clearUponComplete: true,
+  nextSteps: [ step1, step2, step3, step4, step5 ],
 }
 ```
 
-Note: Actions in the action list should not contain functions or anything else not serializable,
-due to the fact that the action list is stored in state.
-
-
-## Action List Lifecycle
-
-### Create
-
-First the list is created. This is usually by helper methods that take in a given
-set of edit states and produce the action list object that is then stored in state.
-After this, the first step of the list can be started.
-
-### Step Start
-
-The next step of the action list is started, the action fired to do the work of this
-step, and then the step object is given a `startTime` field with a timestamp.
-This both tracks timing and denotes the current status of the step as "running".
-
-#### Action
-
-This part is simple. When the step is started, the associated action is dispatched.
-
-#### Asynchronous Response
+As the first step starts, it is designated as the `currentStep`
+```
+{
+  currentStep: step1,
+  nextSteps: [ step2, step3, step4, step5 ],
+}
+```
 
-For asynchronous operations, it's recommended that a data-layer handler be set to
-listen for step actions and perform the asynchronous steps as necessary.
+Then after it completes, it moves to the `prevSteps` array.
+```
+{
+  prevSteps: [ step1 ],
+  currentStep: null,
```

Comment:
Are we using this fact? What is possible with `currentStep: null` that isn't useful without it? I'm just asking because it feels a bit odd to me when we have these nice lists which can't be in an invalid state and then a `null` pops up between them.

what does it mean to have not started the next one? I suppose you are using the `succeeded` and `failed` actions to have the middleware advance this data structure?